api/v1beta1/common_types.go

```diff
@@ -153,6 +163,26 @@ const (
 	VariableDefinitionFromInline = "inline"
 )
 
+// MachineSetPreflightCheck defines a valid machine set preflight check.
+type MachineSetPreflightCheck string
+
+const (
+	// MachineSetPreflightCheckAll can be used to represent all the MachineSet preflight checks.
+	MachineSetPreflightCheckAll MachineSetPreflightCheck = "all"
+
+	// MachineSetPreflightCheckKubeadm is the name of the kubeadm preflight check.
```

Comment:
I think it would be good to document for each check under which circumstances it is executed / what it depends on (e.g. controlPlaneRef must exist in Cluster, ...)